File: src/main/java/com/github/nicomincuzzi/domain/Utensil.java
package com.github.nicomincuzzi.domain;

public class Utensil {

    private final String name;

    public Utensil(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public boolean isSameThan(String item) {
        return item.equalsIgnoreCase(name);
    }
}

File: src/main/kotlin/com/github/nicomincuzzi/maze/Utensil.kt
package com.github.nicomincuzzi.maze

class Utensil(val name: String = "") {

    fun isSameThan(item: String): Boolean {
        return item.equals(name, ignoreCase = true)
    }

    fun isSameThan(item: Utensil): Boolean {
        return item.name.equals(name, ignoreCase = true)
    }

}